multiqc/modules/seqkit/stats.py

```diff
@@ -0,0 +1,394 @@
+"""MultiQC submodule to parse output from seqkit stats"""
+
+import logging
+from typing import Dict
+
+from multiqc import BaseMultiqcModule
+from multiqc.plots import bargraph, table
+
+log = logging.getLogger(__name__)
+
+
+def parse_seqkit_stats(module: BaseMultiqcModule) -> int:
+    """Find seqkit stats logs and parse their data"""
+
+    seqkit_stats: Dict[str, Dict] = {}
+
+    for f in module.find_log_files("seqkit/stats"):
+        parsed_data = parse_stats_report(f["f"])
+        if parsed_data:
+            for s_name, data in parsed_data.items():
+                s_name = module.clean_s_name(s_name, f)
+                if s_name in seqkit_stats:
+                    log.debug(f"Duplicate sample name found! Overwriting: {s_name}")
+                module.add_data_source(f, s_name=s_name, section="stats")
+                seqkit_stats[s_name] = data
+
+    # Filter to strip out ignored sample names
+    seqkit_stats = module.ignore_samples(seqkit_stats)
+
+    if len(seqkit_stats) == 0:
+        return 0
+
+    # Superfluous function call to confirm that it is used in this module
+    # Replace None with actual version if it is available
+    module.add_software_version(None)
+
+    # Check which columns are present in the data
+    has_quality_metrics = any("Q20_pct" in d for d in seqkit_stats.values())
+    has_n50 = any("N50" in d for d in seqkit_stats.values())
+    has_gaps = any("sum_gap" in d for d in seqkit_stats.values())
+
+    # General Stats Table
+    general_stats_headers: Dict = {
+        "num_seqs": {
+            "title": "Sequences",
+            "description": "Number of sequences",
+            "scale": "Blues",
+            "format": "{:,.0f}",
+            "shared_key": "seq_count",
+        },
+        "sum_len": {
+            "title": "Total bp",
+            "description": "Total number of bases",
+            "scale": "Greens",
+            "format": "{:,.0f}",
+            "shared_key": "base_count",
+            "hidden": True,
+        },
+        "avg_len": {
+            "title": "Avg Length",
+            "description": "Average sequence length",
+            "scale": "Purples",
+            "format": "{:,.1f}",
+            "suffix": " bp",
+        },
+        "GC_pct": {
+            "title": "GC%",
+            "description": "GC content percentage",
+            "min": 0,
+            "max": 100,
+            "scale": "RdYlGn",
```

Comment:
This colour scheme is inappropriate, it's used when high numbers (close to 100) are "good".